api-registration/server/index.js

```diff
@@ -0,0 +1,63 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+// const { query } = require('express');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+  const sql = `
+    insert into "users" ("username", "hashedPassword")
+    values ($1, $2)
+    returning "userId", "username", "createdAt"
+  `;
+  argon2.hash(password)
+    .then(res => {
+      const params = [username, res];
+      return db.query(sql, params);
+    })
+    .then(result => {
+      const response = result.rows;
+      res.status(201).json(response);
```

Comment:
Here you are sending back an array with a single object, we only want to send back the object without the array. So you would update line 40:

```JAVASCRIPT
const response = result.rows[0];
```

Reply:
done